src/dotnet/commands/dotnet-build/IncrementalPreconditionManager.cs

```diff
@@ -0,0 +1,117 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.DotNet.Tools.Compiler;
+using Microsoft.DotNet.ProjectModel;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.Cli.Compiler.Common;
+using Microsoft.DotNet.ProjectModel.Utilities;
+
+namespace Microsoft.DotNet.Tools.Build
+{
+    class IncrementalPreconditionManager
```

Comment:
Nit: can you explicitly declare the visibility on these classes? It's #5 in https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md